### PR TITLE
Expanded the courseSpecification to allow custom text searches. Added some controller and integration tests

### PR DESCRIPTION
**Pull request by AndreSoderlund**

### CODE REVIEW

File: src/test/java/se/sundsvall/educationfinder/api/CoursesResourceTest.java [hunk 1]
```diff
@@ -348,4 +352,35 @@ void findFilterValuesForStudyLocation() {
 			.hasSize(5)
 			.containsExactly("HÄRNÖSAND", "KRAMFORS", "SUNDSVALL", "ÖRNSKÖLDSVIK", "ÖSTERSUND");
 	}
+
+	private static Stream<Arguments> queryParameters() {
```
Comment (Cheezi747) [inline src/test/java/se/sundsvall/educationfinder/api/CoursesResourceTest.java:356]: Should probably be in the top of the class.

Comment (AndreSoderlund) [reply src/test/java/se/sundsvall/educationfinder/api/CoursesResourceTest.java:356]: I put it next to the test that is using it. IMO that makes it easier to work with

File: src/integration-test/java/se/sundsvall/educationfinder/apptest/CoursesIT.java [hunk 2]
```diff
@@ -183,4 +183,35 @@ void test17_findAllByLatestApplicationAfterAndLatestApplicationBefore() {
 			.withExpectedResponse(RESPONSE_FILE)
 			.sendRequestAndVerifyResponse();
 	}
+
```
Comment (foobar-1337) [inline src/integration-test/java/se/sundsvall/educationfinder/apptest/CoursesIT.java:186]: Maybe a test that combines the searchString with one of the other filter params, as well?

File: src/test/java/se/sundsvall/educationfinder/api/CoursesResourceTest.java (hunk 1 above)
Comment (Cheezi747) [reply src/test/java/se/sundsvall/educationfinder/api/CoursesResourceTest.java:356]: I see your point. I just think its nice to keep static helpers at the top, so they dont get lost. But do as you feel.

**Review by foobar-1337 — APPROVED** [review]

**Pull request merged**